js/PickerIOS.ios.js

```diff
@@ -52,101 +54,153 @@ type Props = $ReadOnly<{|
   value?: ?(number | string),
   color?: ?ColorValue,
   testID?: ?string,
 |}>;
 
-const PickerIOSItem = (props: ItemProps): null => {
-  return null;
-};
+type CallbackRef<T> = (T) => mixed;
+type ObjectRef<T> = {current: T, ...};
 
-class PickerIOS extends React.Component<Props, State> {
-  _picker: ?ElementRef<RNCPickerIOSType> = null;
+type Ref<T> = CallbackRef<T> | ObjectRef<T>;
 
-  state: State = {
-    selectedIndex: 0,
-    items: [],
-  };
+/**
+ * Constructs a new ref that forwards new values to each of the given refs. The
+ * given refs will always be invoked in the order that they are supplied.
+ *
+ * WARNING: A known problem of merging refs using this approach is that if any
+ * of the given refs change, the returned callback ref will also be changed. If
+ * the returned callback ref is supplied as a `ref` to a React element, this may
+ * lead to problems with the given refs being invoked more times than desired.
+ */
+function useMergeRefs<T>(...refs: $ReadOnlyArray<?Ref<T>>): CallbackRef<T> {
+  return React.useCallback(
+    (current: T) => {
+      for (const ref of refs) {
+        if (ref != null) {
+          if (typeof ref === 'function') {
+            ref(current);
+          } else {
+            ref.current = current;
+          }
+        }
+      }
+    },
+    [...refs], // eslint-disable-line react-hooks/exhaustive-deps
+  );
+}
 
-  static Item: typeof PickerIOSItem = PickerIOSItem;
+const PickerIOSItem: RNCPickerIOSItemType = (props: ItemProps): null => {
+  return null;
+};
 
-  static getDerivedStateFromProps(props: Props): State {
+const PickerIOSWithForwardedRef: React.AbstractComponent<
+  Props,
+  React.ElementRef<typeof RNCPickerNativeComponent>,
+> = React.forwardRef(function PickerIOS(props, forwardedRef): React.Node {
+  const {
+    children,
+    selectedValue,
+    selectionColor,
+    themeVariant,
+    testID,
+    itemStyle,
+    numberOfLines,
+    onChange,
+    onValueChange,
+    style,
+  } = props;
+
+  const nativePickerRef = React.useRef<React.ElementRef<
+    typeof RNCPickerNativeComponent,
+  > | null>(null);
+
+  const ref = useMergeRefs(nativePickerRef, forwardedRef);
+
+  const [nativeSelectedIndex, setNativeSelectedIndex] = React.useState({
+    value: null,
+  });
+
+  const [items, selectedIndex] = React.useMemo(() => {
+    // eslint-disable-next-line no-shadow
     let selectedIndex = 0;
-    const items = [];
-    React.Children.toArray(props.children).forEach(function (child, index) {
-      if (child.props.value === props.selectedValue) {
+    // eslint-disable-next-line no-shadow
+    const items = React.Children.toArray(children).map((child, index) => {
+      if (child === null) {
+        return null;
+      }
+      if (String(child.props.value) === String(selectedValue)) {
         selectedIndex = index;
       }
-      items.push({
-        value: child.props.value,
-        label: child.props.label,
+      return {
+        value: String(child.props.value),
```

Comment:
this change caused a regression in my app, as the expected type returned by the `onValueChange` handler used to be an integer (if the item's value was initially an int).